pwiz_tools/Skyline/Model/AuditLog/AuditLogEntry.cs

```diff
@@ -206,10 +206,31 @@ public static bool ReadFromFile(string fileName, out string loggedSkylineDocumen
             {
                 using (var alert = new AlertDlg())
```

Comment:
You should throw an exception instead of putting up an alert here.
This code gets called from both the commandline and from the user interface, and the commandline should never bring up a message box.
This exception can be caught in "SkylineWindow.OpenFile" and you can use the usual MessageDlg.ShowWithException to show the error in the usual way.

Reply:
Well, the dialog was there to start with, I just modified the way the message was displayed. But it's a good point. I considered just throwing the exception to catch is downstream, but the catch (no pun intended)  is that we still want to open the Skyline doc even if the audit log is invalid. And if this exception would propagate any further the loading of the document file is canceled.
Possible option would be to test if application has UI and just send the message into System.out if it has none.